### PR TITLE
proxy-config injection from cluster-apps

### PR DESCRIPTION
**Pull request by bavarianbidi**
* The `cluster-values` secret which is newly created by `cluster-apps-operator` (open PR https://github.com/giantswarm/cluster-apps-operator/pull/290) contains a valid `containerd` proxy configuration.
* CAPI is waiting, with machine-creation, once the secret got created by `cluster-apps-operator`.
* This currently work with this (https://github.com/giantswarm/cluster-api-provider-cloud-director/pull/1) forked version of `capvcd`.

This PR:

- adds/changes/removes etc

### Testing

Description on how {APP-NAME} can be tested.

- [ ] fresh install works
- [ ] upgrade from previous version works

#### Other testing

Description of features to additionally test for {APP-NAME} installations.

- [ ] check reconciliation of existing resources after upgrading
- [ ] X still works after upgrade
- [ ] Y is installed correctly

### Checklist

- [ ] Update changelog in CHANGELOG.md.
- [ ] Make sure `values.yaml` and `values.schema.json` are valid.
- [ ] Update Lastpass values if required.

### CODE REVIEW

**Comment by vxav** [comment]
Nice, I like that it's in a secret, maybe we should get in touch with CAPA guys in kaas sync to align with this https://github.com/giantswarm/cluster-aws/blob/master/helm/cluster-aws/templates/_helpers.tpl#L68

**Comment by bavarianbidi** [comment]
> Nice, I like that it's in a secret, maybe we should get in touch with CAPA guys in kaas sync to align their implementation with this [giantswarm/cluster-aws@`master`/helm/cluster-aws/templates/_helpers.tpl#L68](https://github.com/giantswarm/cluster-aws/blob/master/helm/cluster-aws/templates/_helpers.tpl?rgh-link-date=2022-11-09T10%3A23%3A23Z#L68)

I'm already in contact with Vaclav about this. To bring it in kaas-sync i want to document the current implementation (and the assumptions behind) - otherwise it might be hard to follow the discussion

File: helm/cluster-cloud-director/templates/_helpers.tpl [hunk 1]
```diff
@@ -19,6 +19,14 @@ Create chart name and version as used by the chart label.
 infrastructure.cluster.x-k8s.io/v1beta1
 {{- end -}}
 
+
+{{- define "proxyFiles" -}}
```
Comment (erkanerol) [inline helm/cluster-cloud-director/templates/_helpers.tpl:23]: Where is this used? I couldn't find.

**Review by erkanerol — APPROVED** [review]
LGTM

File: helm/cluster-cloud-director/values.yaml [hunk 2]
```diff
@@ -44,10 +44,8 @@ network:
     vipSubnet: ""  # Virtual IP CIDR for the external network.
 
 proxy:  # (Optional) Defines HTTP proxy environment variables.
+  secretName: ""
   enabled: false
-  httpProxy: ""
-  httpsProxy: ""
-  noProxy: ""  # Bypass the proxy for specific hosts - cluster IP range for instance.
```
Comment (kopiczko) [inline helm/cluster-cloud-director/values.yaml:50]: If we want to deliver it like that here I would like it this secret to be documented in the repo and remove all references of `cluster-apps-operator` in the PR. This is a public repo and can be used by anyone in the wild. We should not make it confusing for potential users. For the GS platform we can document the inner working in our official docs.

**Comment by kopiczko** [comment]
I'm not a fan of the approach. I expressed my concerns here https://github.com/giantswarm/cluster-apps-operator/pull/290#issuecomment-1313444808

File: helm/cluster-cloud-director/values.yaml (hunk 2 above)
Comment (bavarianbidi) [reply helm/cluster-cloud-director/values.yaml:50]: @kopiczko 

> If we want to deliver it like that here I would like it this secret to be documented in the repo

with documentation you mean a comment at the end of the line? If so, will do it that way as i didn't find any other "documentation" in this repo where the values are documented.

> This is a public repo and can be used by anyone in the wild. 

Unfourtanetly we have multiple other giantswarm helm charts where we refer to values from `cluster-apps-operator`/`cluster-operator` where it's not that easy for potential customers to re-use the helm-chart.
If this is still the goal for all our repos (what i assume) i guess we hit this topic already in the past (as other apps are using values from `cluster-apps-operator`/`cluster-operator` for months/years) i'm wondering if there was a decission how to solve this topic in general?!

Comment (kopiczko) [reply helm/cluster-cloud-director/values.yaml:50]: > Unfourtanetly we have multiple other giantswarm helm charts where we refer to values from `cluster-apps-operator`/`cluster-operator` where it's not that easy for potential customers to re-use the helm-chart.

Is this a reason to make yet another chart like that? We already have a fork of `cluster-openstack` [in the wild](https://github.com/eschercloudai/cluster-openstack) so it isn't exactly true. 

BTW isn't `<clusterName>-cluster-values` linked in the cluster App already? Maybe we can stick with the previous values schema?
